docs(developer.md): add note about ulimit workaround (#1171)
qri-io/qri: DEVELOPERS.md

[yarn-install]: https://yarnpkg.com/en/docs/install

## <a name="troubleshooting"> Troubleshooting

> #### "Too Many Open Files" on OS X

Reviewer: Pedantic fix (not that anybody is confused by this, though):

```suggestion
#### "Too Many Open Files" on MacOS
```

> If while running the test suite you get failing tests with "too many open files", you may need to adjust the number of files your shell process is willing to open. This is particularly common with the `p2p` and `cmd` pacakges. Both rely heavily on interacting temporary directories on the file system to run tests.

Reviewer: ```suggestion
If, while running the test suite, you get failing tests with "too many open files", you may need to adjust the number of files your shell process is willing to open. This is particularly common with the `p2p` and `cmd` pacakges. Both rely heavily on interacting temporary directories on the file system to run tests.
```

> Often this is caused by having too low a `ulimit` value for the shell. Try the following command:

Reviewer: ```suggestion
Often this is caused by having too low a file limit for your shell. You can use the [`ulimit` command](https://ss64.com/osx/ulimit.html) to check or change the limits. Try the following command to set the limit to 1000 open files:
```

```
ulimit -S -n 1000
```

> We recommend a ulimit value of at least 1000, feel free to go higher if you need. Some other software require values as high as 10000 to avoid max open file issue.

Reviewer: ```suggestion
We recommend a ulimit value of at least 1000; feel free to go higher if you need. Some other software require values as high as 10000 to avoid the max open file issue.
```
